tracer/test/Datadog.Trace.Tests/Configuration/ConfigurationSourceTests.cs

```diff
@@ -185,13 +185,19 @@ public void DefaultSetting(Func<TracerSettings, object> settingGetter, object ex
             }
             else if (key == ConfigurationKeys.AgentHost || key == ConfigurationKeys.AgentPort)
             {
-                // We need to ensure DD_TRACE_AGENT_URL is empty.
+                // We need to ensure all the agent URLs are empty.
+                string originalHost = Environment.GetEnvironmentVariable(ConfigurationKeys.AgentHost);
```

Comment:
Could be a good use-case for an IDisposable pattern?

Reply:
_Theoretically_ yes. We could save and restore _all_ env variables before and after the test. That's probably the right thing to do...